Source/WebGPU/WebGPU/Pipeline.h

```diff
@@ -35,6 +35,6 @@ std::optional<LibraryCreationResult> createLibrary(id<MTLDevice>, const ShaderMo
 
 MTLFunctionConstantValues *createConstantValues(uint32_t constantCount, const WGPUConstantEntry* constants, const WGSL::Reflection::EntryPointInformation&);
 
-id<MTLFunction> createFunction(id<MTLLibrary>, const WGSL::Reflection::EntryPointInformation&, const WGPUProgrammableStageDescriptor*, NSString *label);
+id<MTLFunction> createFunction(id<MTLLibrary>, const WGSL::Reflection::EntryPointInformation&, unsigned constantCount, const WGPUConstantEntry*, NSString *label);
```

Comment:
I don't understand this change. this function is called once for the vertex shader and once for the fragment shader. The WGPUProgrammableStageDescriptor holds the `WGPUConstantEntry const * constants` for that shader. This change shouldn't be necessary.

Reply:
`WGPUProgrammableStageDescriptor` is not part of `WGPURenderPipelineDescriptor`. I think it is to unify behavior between the render and compute pipelines

Reply:
one could cast a `WGPUVertexState*` or `WGPUFragmentState*` to a `WGPUProgrammableStageDescriptor*` and have that work based on the struct definitions but that is quite fragile and depends on matching alignments of the structs and no change in member ordering